src/main/java/Duke.java

```diff
@@ -1,10 +1,94 @@
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class Duke {
+    private static void welcomeMessage() {
+        String greetings = "Hello! I'm Duke\n"
+                + "What can I do for you?\n";
+        System.out.println(greetings);
+    }
+
+    private static void goodbyeMessage() {
+        String goodbyeMessage = "Bye. Hope to see you again soon!";
+        System.out.println(goodbyeMessage);
+    }
+
+    public static void addTaskToTaskList(String userInput, int taskNumber, Task[] taskList) {
+        String[] userInputSplitArray = userInput.split(" ");
+        String taskType = userInputSplitArray[0];
+
+        String taskDescription = "";
+        for (int i = 1; i < userInputSplitArray.length; i++) {
+            taskDescription = taskDescription + " " + userInputSplitArray[i];
+        }
+        taskDescription = taskDescription.stripLeading();
+
+        if (taskType.equals("todo")) {
+            Todo todo = new Todo(taskDescription);
+            taskList[taskNumber] = todo;
+        } else if (taskType.equals("deadline")) {
+            String[] deadlineSplitByDate = taskDescription.split("/by");
+            String deadlineTask = deadlineSplitByDate[0];
+            String deadlineDate = deadlineSplitByDate[1];
+            Deadline deadline = new Deadline(deadlineTask, deadlineDate);
+            taskList[taskNumber] = deadline;
+        } else if (taskType.equals("event")) {
+            String[] eventSplitByDate = taskDescription.split("/at");
+            String eventTask = eventSplitByDate[0];
+            String eventDate = eventSplitByDate[1];
+            Event event = new Event(eventTask, eventDate);
+            taskList[taskNumber] = event;
+        }
+
+        System.out.println("Now you have " + taskNumber + " tasks in the list.\n");
+    }
+
+    public static void markTaskAsDone(String userInput, Task[] taskList) {
+        String[] userInputArray = userInput.split(" ");
+        int doneTaskNumber = Integer.parseInt(userInputArray[1]);
+        Task taskDone = taskList[doneTaskNumber];
```

Comment:
I recall seeing a `boolean doneTask` elsewhere in the code - this could get confusing quickly, especially given that they're different types too. Perhaps one way to avoid is to change the name of the `boolean` variable.